Sage/carbon
# docs(badge): ensure demos do not render partially offscreen FE-5231

fix #5265
fix #5276
fix #5274

### Proposed behaviour


Badge demos:
Adds margin to wrapper to ensure all demos do not render offscreen, refactors test story to be tsx file instead of mdx

![image](https://user-images.githubusercontent.com/44157880/176918645-3b79d49c-99cb-430f-9822-6c5d201c0597.png)


FlatTable demo:
Updates the value passed to `FlatTableRowHeader` `width` prop so it is a number instead of string

Button demos:
Adds a margin top to all buttons to ensure they are spaced consistently even when they wrap at
smaller screen sizes

![image](https://user-images.githubusercontent.com/44157880/176918587-add3af5c-520c-432b-9f1b-18f8cd838738.png)


### Current behaviour


The Badge stories are rendered partially off the screen
There is a prop type warning in the FlatTable expandable stories
The Buttons overlap each other when the demos wrap in smaller viewports
 
### Checklist




#### QA


### Additional context



### Testing instructions


See linked issues, can be tested in the relevant component stories

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
